### PR TITLE
Fix server_id bug

### PR DESCRIPTION
**Pull request by osheroff**
maxwell had a bug in which schemas were always saved with server_id == 1 -- meaning someone could switch server_ids out from underneath us without us knowing.  bad news!  this fixes.

note that this means if the master flips, we will end up re-capturing the schema.  that's ok for now.

@zendesk/rules

### CODE REVIEW

File: src/main/java/com/zendesk/maxwell/schema/SchemaStore.java [hunk 1]
```diff
@@ -204,12 +210,23 @@ public static SchemaStore restore(Connection connection,
...
 		PreparedStatement p;
-		ResultSet schemaRS = findSchema(targetPosition);
+		boolean shouldResave = false;
+		ResultSet schemaRS = findSchema(targetPosition, this.serverID);
+
+		if (schemaRS == null) {
+			// old versions of Maxwell had a bug where they set every server_id to 1.
+			// try to upgrade.
+
+			LOGGER.info("found schema with server_id == 1, re-saving...");
```
Comment (vanchi-zendesk) [inline src/main/java/com/zendesk/maxwell/schema/SchemaStore.java]: A new server will never have it as 1 right?

Comment (vanchi-zendesk) [inline src/main/java/com/zendesk/maxwell/schema/SchemaStore.java]: oh. nm. dumb question.

**Comment by vanchi-zendesk** [comment]
:+1:

File: src/main/java/com/zendesk/maxwell/schema/SchemaStore.java (hunk 1 above)
Comment (osheroff) [inline src/main/java/com/zendesk/maxwell/schema/SchemaStore.java]: it could.  that would trigger a re-save every time.  harumph.

Comment (vanchi-zendesk) [inline src/main/java/com/zendesk/maxwell/schema/SchemaStore.java]: er. I thought in that case schemaRS will not be null, no?

Comment (vanchi-zendesk) [inline src/main/java/com/zendesk/maxwell/schema/SchemaStore.java]: I mean the second time around?

Comment (osheroff) [inline src/main/java/com/zendesk/maxwell/schema/SchemaStore.java]: oh.  right!  yeah.  it'll have been accidentally correct.

**Pull request merged**